contrib/src/main/java/org/archive/modules/extractor/ExtractorYoutubeDL.java

```diff
@@ -581,6 +610,22 @@ public WARCRecordInfo buildRecord(CrawlURI curi, URI concurrentTo)
         recordInfo.setMimetype("application/vnd.youtube-dl_formats+json;charset=utf-8");
         recordInfo.setEnforceLength(true);
 
+        //Use the recorder object to calculate the content digest and store it on the curi.
+        //Must be calculated now, before the warc writer closes the file stream.
+        //We don't need an extra copy, so just write to NullOutputStream.
+        String recorderBaseName = "ExtractorYoutubeDL-" + nextRecorderId.getAndIncrement();
+        Recorder recorder = new Recorder(new File(controller.getScratchDir().getFile(), recorderBaseName),
+                controller.getRecorderOutBufferBytes(), controller.getRecorderInBufferBytes());
+        recorder.getRecordedInput().setDigest("sha1");
+        getLocalTempFile().seek(0);
+        recorder.inputWrap(Channels.newInputStream(getLocalTempFile().getChannel()));
+        recorder.getRecordedInput().startDigest();
+        recorder.outputWrap(new NullOutputStream());
+        recorder.getRecordedInput().readFully();
+        curi.getData().put(YDL_JSON_FILE_DIGEST,recorder.getRecordedInput().getDigestValue());
+        recorder.getRecordedOutput().close();
```

Comment:
Since we don't call recorder.cleanup() would this leak temp files when the resource is larger than the buffer size and Recorder spills to disk?

Are we only using Recorder here for the side-effect of calculating the digest? Since we have DigestUtils from Commons Codec available maybe we could just do:

```java
getLocalTempFile().seek(0);
InputStream inputStream = Channels.newInputStream(getLocalTempFile().getChannel());
curi.getData().put(YDL_JSON_FILE_DIGEST, DigestUtils.sha1(inputStream));
```

Reply:
Yeah, my intention was to reuse existing code paths when possible, but this would simplify things, and sha1 should be pretty safe. Thanks for taking a look.

Reply:
The hashing portion has been refactored. Much simpler now.